FSIM_USAGE.md

```diff
@@ -0,0 +1,236 @@
+- Commands run with the privileges of the device onboard process
+- Only predefined safe commands are supported
+- Command output is captured and logged
+- Device must explicitly enable command execution with `--echo-commands` for security
+
+## fdo.download FSIM
+
+### Purpose
+Transfer files from the owner server to the device during onboarding.
+
+### Usage
+```bash
+go-fdo-server owner 127.0.0.1:8043 \
+  --db /tmp/fdo/db/own.db --db-pass "$DB_PASS" \
+  --device-ca-cert /tmp/fdo/keys/device_ca_cert.pem \
+  --owner-key /tmp/fdo/keys/owner_key.der \
+  --command-download /path/to/local/file1.txt \
+  --command-download /path/to/local/file2.conf
+```
+
+### Parameters
+- `--command-download <file_path>`: Specify a local file path to transfer to the device
+- Flag can be used multiple times to transfer multiple files
+
+### Device-Side Requirements
+The device must specify a directory for downloaded files using the `--download` option:
+```bash
+go-fdo-client onboard --key ec256 --kex ECDH256 --debug --blob /tmp/fdo/cred.bin --download /tmp/downloads
+```
+
+### File Transfer Details
+- Files are read from the owner server's local filesystem
+- Downloaded files are named using the same name given in the `--command-download` filepath (e.g., "file1.txt", "file2.conf")
+- Transfer is mandatory (MustDownload: true)
+
+### Example
+```bash
+# Prepare files to download
+echo "Device configuration data" > /tmp/device-config.txt
+echo "Application settings" > /tmp/app-settings.json
+
+# Start owner with download FSIM
+go-fdo-server owner 127.0.0.1:8043 \
+  --db /tmp/fdo/db/own.db --db-pass "$DB_PASS" \
+  --device-ca-cert /tmp/fdo/keys/device_ca_cert.pem \
+  --owner-key /tmp/fdo/keys/owner_key.der \
+  --command-download /tmp/device-config.txt \
+  --command-download /tmp/app-settings.json
+```
+
+### Security Considerations
+- Download directory permissions should be appropriately configured
+- Validate downloaded file contents and integrity after transfer
+
+### Error Handling
+- If a specified file cannot be opened, the owner service will log a fatal error
+- File permissions on the owner server must allow reading by the service process
+
+## fdo.upload FSIM
+
+### Purpose
+Transfer files from the device to the owner server during onboarding.
+
+### Usage
+```bash
+go-fdo-server owner 127.0.0.1:8043 \
+  --db /tmp/fdo/db/own.db --db-pass "$DB_PASS" \
+  --device-ca-cert /tmp/fdo/keys/device_ca_cert.pem \
+  --owner-key /tmp/fdo/keys/owner_key.der \
+  --upload-directory /tmp/uploads \
+  --command-upload device-logs.txt \
+  --command-upload system-info.json
+```
+
+### Parameters
+- `--upload-directory <dir_path>`: Directory on owner server where uploaded files will be stored
+- `--command-upload <filename>`: Name of file to request from device
+- Upload flag can be used multiple times for multiple files
+
+### Device-Side Requirements
+The device must specify directories containing uploadable files using the `--upload` parameter:
```

Comment:
the directory path provided via --upload has is treated as relative path

Reply:
Updated